src/mococrw/hash.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "mococrw/openssl_wrap.h"
+
+namespace mococrw
+{
+
+void sha256(std::string message, std::string *outDigest);
```

Comment:
I am ok with these convenience functions. However, why don't they just return the digest? Moreover, I have the impression that it is strange that they sometimes return a string an sometimes raw bytes. We no have the convenience hex conversion function. Hence, I would opt for always return bytes (in form of a memory managed vector)

Reply:
switched all functions to return a vector